public/app.js

```diff
@@ -0,0 +1,101 @@
+var app = angular.module('libraryApp', ['ngRoute', 'ngResource']);
+
+////////////
+// ROUTES //
+////////////
+
+app.config(['$routeProvider', '$locationProvider', function ($routeProvider, $locationProvider) {
+  $routeProvider
+    .when('/', {
+      templateUrl: 'templates/books/index.html',
+      controller: 'BooksIndexCtrl'
+    })
+    .when('/books/:id', {
+      templateUrl: 'templates/books/show.html',
+      controller: 'BooksShowCtrl'
+    });
+    $locationProvider
+    .html5Mode({
+      enabled: true,
+      requireBase: false
+    });
+}]);
+
+/////////////////
+// API CALL //
+/////////////////
+
+app.factory("Book", ["$resource", function($resource) {
+  return $resource('https://api.parse.com/1/classes/Book/:bookId', {bookId: "@bookId"}, {
+    query: {
+      method: 'GET',
+      headers: {
+      'X-Parse-Application-Id': 'JjfvfpGDvmx9etMgv916ittvni3QkdvdXY6ODbEc',
+      'X-Parse-REST-API-Key': 'NaebdwatVq5Evg2VPBGhLyOF1moLiHqnCSvUcn72'
+    }
+    },
+    save: {
+      method: 'POST',
+      headers: {
+      'X-Parse-Application-Id': 'JjfvfpGDvmx9etMgv916ittvni3QkdvdXY6ODbEc',
+      'X-Parse-REST-API-Key': 'NaebdwatVq5Evg2VPBGhLyOF1moLiHqnCSvUcn72'
+    }
+    },
+    update: { 
+      method:'PUT',
+      headers: {
+      'X-Parse-Application-Id': 'JjfvfpGDvmx9etMgv916ittvni3QkdvdXY6ODbEc',
+      'X-Parse-REST-API-Key': 'NaebdwatVq5Evg2VPBGhLyOF1moLiHqnCSvUcn72'
+    }
+    },
+    delete: { 
+      method:'DELETE',
+      headers: {
+      'X-Parse-Application-Id': 'JjfvfpGDvmx9etMgv916ittvni3QkdvdXY6ODbEc',
+      'X-Parse-REST-API-Key': 'NaebdwatVq5Evg2VPBGhLyOF1moLiHqnCSvUcn72'
+    }
+    }
+  });
+}]);
+
+/////////////////
+// CONTROLLERS //
+/////////////////
+
+app.controller("BooksIndexCtrl", ["$scope", "Book", function ($scope, Book) {
+  Book.query().$promise
```

Comment:
Checkout some examples of `query` to see if you can reduce some of this code: https://docs.angularjs.org/api/ngResource/service/$resource